pkg/k8s/apis/cilium.io/utils/utils.go

```diff
@@ -93,8 +93,17 @@ func getEndpointSelector(namespace string, labelSelector *slim_metav1.LabelSelec
 	// Those pods don't have any labels, so they don't have a namespace label either.
 	// Don't add a namespace label to those endpoint selectors, or we wouldn't be
 	// able to match on those pods.
-	if !matchesInit && !es.HasKey(podPrefixLbl) && !es.HasKey(podAnyPrefixLbl) && namespace != "" {
-		es.AddMatch(podPrefixLbl, namespace)
+	if !matchesInit && !es.HasKey(podPrefixLbl) && !es.HasKey(podAnyPrefixLbl) {
+		if namespace != "" {
```

Comment:
nit: We will usually put the == case first so that when trying to interpret the `else` case we don't have to do the mental swap involved in  `! (namespace != "")`.